src/Plugin/media/Source/MediaAvPortalPhotoSource.php

```diff
@@ -46,4 +64,32 @@ public function getMetadata(MediaInterface $media, $name) {
     return parent::getMetadata($media, $name);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function transformUrlToReference(array $url): ?string {
+    preg_match('/(\d+)/', $url['query']['ref'], $matches);
+    // The reference should be in the format P-xxxx-00-yy where xxxx and
+    // yy are numbers.
+    // Sometimes no dash is present, so we have to normalise the reference
+    // back.
+    if (isset($matches[0])) {
+      return 'P-' . $matches[0] . '/00-' . sprintf('%02d', $url['fragment'] + 1);
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transformReferenceToUrl(string $reference): ?string {
```

Comment:
This should return a string in all cases, even if no matches are found.

Also, no need for empty line at the beginning of the method.